pkgs/by-name/sw/swhkd/package.nix

```diff
@@ -0,0 +1,36 @@
+{lib, rustPlatform, fetchFromGitHub, pkg-config, pkgs }:
+
+  rustPlatform.buildRustPackage rec {
+    pname = "swhkd";
+    version = "1.2.1";
+
+    src = fetchFromGitHub {
+        owner = "waycrate";
+        repo = "swhkd";
+        rev = "refs/tags/${version}";
+        hash = "sha256-VQW01j2RxhLUx59LAopZEdA7TyZBsJrF1Ym3LumvFqA=";
+      };
+
+    nativeBuildInputs = with pkgs; [
+      pkg-config
+    ];
+
+    cargoHash = "sha256-NAVqwYJA0+X0dFC3PBaW+QJxvJtSgl4Y/VNfNO3jnLA=";
+
+    preBuild = ''
+      cargo generate-lockfile
+    '';
```

Comment:
what's this for? the upstream repository already has a lockfile, and we shouldn't modify that, otherwise our builds may not be reproducible.

Reply:
> what's this for? the upstream repository already has a lockfile, and we shouldn't modify that, otherwise our builds may not be reproducible.

Hello, excuse me for the late response. I had been away over Easter. 
The reason these lines are there is because I had been getting these error codes:
`
> error: the lock file /build/source/Cargo.lock needs to be updated but --frozen was passed to prevent this
> If you want to try to generate the lock file without accessing the network, remove the --frozen flag and use --offline instead.
`
Generating a new lockfile helped the build pass.

Reply:
cargo passes --locked to enforce reproducibility.  instead of subverting that, if the upstream has a missing or broken lockfile, the correct solution is to vendor a fixed version of the lockfile within nixpkgs, then set `cargoLock.lockFile` to the vendored version.

if cargo continues to complain, use `ln -sf ${cargoLock.lockFile} Cargo.lock` in `postPatch` to make sure the upstream lockfile is replaced with the vendored one.

Reply:
Hello, sorry it's taking so long. I could not figure out how to do what you suggested. 

However, I did in fact realize that this version of swhkd is considerably old. This package build currently points to a relatively old version released in 2022. 

I would just help PR a release into swhkd, but there is currently a problem with the most recent version of the actual program which is not desirable at all. Currently as it stands, when the program is activated with pkexec, it absolutely kills the WiFi until the network manager is restarted. This is something to do with rfkill. 

I'm just waiting for this to be resolved so there will be an all clear for there to be a new release.